source/Public/New-GuestConfigurationPolicy.ps1

```diff
@@ -275,26 +296,39 @@ function New-GuestConfigurationPolicy
 
     # Download package
     $tempPath = Reset-GCWorkerTempDirectory
+    $packagePath = Join-Path -Path $tempPath -ChildPath 'extracted'
 
-    $packageFileDownloadName = 'temp.zip'
-    $packageFileDownloadPath = Join-Path -Path $tempPath -ChildPath $packageFileDownloadName
-
-    if (Test-Path -Path $packageFileDownloadPath)
+    if ($ContentManagedIdentity -and $ContentPath)
     {
-        $null = Remove-Item -Path $packageFileDownloadPath -Force
-    }
+        $contentHash = (Get-FileHash -Path $ContentPath -Algorithm 'SHA256').Hash
 
-    $null = Invoke-WebRequest -Uri $ContentUri -OutFile $packageFileDownloadPath
+        $packagePresentIn = $ContentPath
```

Comment:
You don't need this extra $packagePresentIn variable.
You can set $packageFileDownloadPath to ContentPath here
Then please move the contentHash extraction after the if statement.
There's no need to have the same code for it in 2 places.

Reply:
Changed the code accordingly.